openpgp/packet/private_key.go

```diff
@@ -221,21 +222,23 @@ func (pk *PrivateKey) parse(r io.Reader) (err error) {
 		if !pk.Encrypted {
 			count = count+2 /* two octet checksum */
 		}
-		pk.encryptedData = make([]byte, count)
-		_, err = readFull(r, pk.encryptedData)
+		privateKeyData = make([]byte, count)
+		_, err = readFull(r, privateKeyData)
 		if err != nil {
 			return
 		}
 	} else {
-		pk.encryptedData, err = ioutil.ReadAll(r)
+		privateKeyData, err = ioutil.ReadAll(r)
 		if err != nil {
 			return
 		}
 	}
 	if !pk.Encrypted {
-		return pk.parsePrivateKey(pk.encryptedData)
+		pk.encryptedData = nil
```

Comment:
I'm pretty sure it's already nil at this point, I don't think you need to set this here

Reply:
I checked, fair enough, applied.
I don't love relying on implicit defaults but OK.